### DISCUSSION TITLE
In 2023 kcp.....

### DISCUSSION THREAD

**Discussion by pweil-** [kcp-dev/kcp#2454] [Ideas]
### 🔮 Let's chat about the future of kcp! 🔮 

In this discussion we'd like to hear your thoughts in the form of statements like:

* In 2023 kcp should.....
* We'd be crazy if we didn't ..... in 2023

Areas of improvement, predictions, things you'd advocate for from a developer, customer, community, or business perspective, or other thoughts you have are all welcome. We'll try to chat about these in the December 13, 2022 community call.  If we remember, we can use it as a look back at the end of 2023.

**Comment by fgiloux** [comment]
In 2023 there should be an easy way to take an existing operator and to turn it into a service offered on kcp, able to scale to millions of managed resources over multiple regions.
An easy way would mean: no substantial code change, ideally not more than swapping a couple of library dependencies and running some code generation. The result should also be able to run on plain Kubernetes.

**Comment by maleck13** [comment]
In 2023 kcp should be able to provide a valuable multi-cluster workload service that delivers on the clear value of transparent multi cluster. It should hold true to the idea that I can come with my existing application and, with as few changes as possible, deploy it to one or more clusters and have it "just work"

**Comment by sttts** [comment]
In 2023, 
- sharding should be ready and totally boring and taken for granted
- our programming models and APIs should be promoted to post-alpha levels
- the repository should be be split in kcp-core, kcp-tenancy, kcp-tmc, kcp-edge or similar, and kcp-core will see adoption by other parties not yet involved in kcp today, with totally different use-cases, and super excited about how kcp can extend the reach of kube-like APIs.

**Comment by kasturinarra** [comment]
In 2023,

1.    Documentation added to every feature that is being checked in and an example of how it should be used.
2.    Clarity to the end user on what functionality is complete and what is in progress at a glance.
3.    Document any changes/ UX changes that will effect end user in the upcoming release
4.    Any known issues list in the release

**Comment by heiko-braun** [comment]
In 2023 there should be an easy way see what capabilities (Services, APIs?) are available in the KCP ecosystem (CPS?) that I can leverage on my dataplane. Me, being an novice user, would like to describe which capabilities I need and have those APIs available in my workspace in a seamless fashion.
